### PR TITLE
Feature/unit test create resource

### PR DESCRIPTION
**Pull request by jbout**
Requires https://github.com/oat-sa/extension-tao-lti/pull/199

### CODE REVIEW

File: config/header/uriProvider.conf.php [hunk 1]
```diff
@@ -12,5 +12,5 @@
  *     oat\generis\model\kernel\uri\MicrotimeRandUriProvider
  *     oat\generis\model\kernel\uri\Bin2HexUriProvider
  * 
- * @see core_kernel_uri_UriService
+ * @see oat\generis\model\kernel\uri\UriProvider
  */
```
Comment (julien-sebire) [inline config/header/uriProvider.conf.php]: Missing newline at end of file.

File: core/kernel/classes/class.Resource.php [hunk 2]
```diff
@@ -576,7 +576,6 @@ public function getUsedLanguages( core_kernel_classes_Property $property)
      */
     public function duplicate($excludedProperties = array())
     {
-        $returnValue = null;
         $returnValue = $this->getImplementation()->duplicate($this, $excludedProperties);
```
Comment (julien-sebire) [inline core/kernel/classes/class.Resource.php]: Please return directly without a temp variable ! :-)

File: core/kernel/persistence/smoothsql/class.Class.php [hunk 3]
```diff
@@ -45,7 +46,7 @@ public function getSubClasses( core_kernel_classes_Class $resource, $recursive =
             $sqlQuery = 'SELECT subject FROM statements WHERE predicate = ? and '.$this->getPersistence()->getPlatForm()->getObjectTypeCondition() .' = ?';
             $sqlResult = $this->getPersistence()->query($sqlQuery, array(OntologyRdfs::RDFS_SUBCLASSOF, $resource->getUri()));
             while ($row = $sqlResult->fetch()) {
-                $returnValue[$row['subject']] = new core_kernel_classes_Class($row['subject']);
+                $returnValue[$row['subject']] = $this->getModel()->getClass($row['subject']);
             }
             return (array) $returnValue;
```
Comment (julien-sebire) [inline core/kernel/persistence/smoothsql/class.Class.php]: Not necessary casting to array.

File: core/kernel/persistence/smoothsql/class.Class.php [hunk 4]
```diff
@@ -256,7 +257,7 @@ public function createInstance( core_kernel_classes_Class $resource, $label = ''
 
     	$subject = '';
     	if ($uri == ''){
-			$subject = common_Utils::getNewUri();
+            $subject = $this->getServiceLocator()->get(UriProvider::SERVICE_ID)->provide();
```
Comment (julien-sebire) [inline core/kernel/persistence/smoothsql/class.Class.php]: Nice to have the indentation restored ! ;-)

File: config/header/uriProvider.conf.php (hunk 1 above)
Comment (jbout) [reply config/header/uriProvider.conf.php]: will not add new line in header template file

File: core/kernel/classes/class.Resource.php (hunk 2 above)
Comment (jbout) [reply core/kernel/classes/class.Resource.php]: kept this in place as we are likely to add an event in between in the future

File: manifest.php [hunk 5]
```diff
@@ -32,7 +32,7 @@
     'label' => 'Generis Core',
     'description' => 'Core extension, provide the low level framework and an API to manage ontologies',
     'license' => 'GPL-2.0',
-    'version' => '12.3.2',
```
Comment (siwane) [inline manifest.php]: IT should be a bump of major version, isn't it?

File: core/kernel/classes/class.Resource.php (hunk 2 above)
Comment (julien-sebire) [reply core/kernel/classes/class.Resource.php]: Ok for me then.

**Pull request merged**